### PR TITLE
refactor: update `blas/ext/base/sapx` to follow current project conventions

### LINKED ISSUE #1152: Refactor and update existing extended base BLAS packages according to current project conventions
This should follow the same process as outlined in https://github.com/stdlib-js/stdlib/issues/788 but for `@stdlib/blas/ext/base/*`.

Packages needing updating:

- [x] `@stdlib/blas/ext/base/dapx`
- [x] `@stdlib/blas/ext/base/dapxsum`
- [x] `@stdlib/blas/ext/base/dapxsumkbn`
- [x] `@stdlib/blas/ext/base/dapxsumkbn2`
- [x] `@stdlib/blas/ext/base/dapxsumors`
- [x] `@stdlib/blas/ext/base/dapxsumpw`
- [x] `@stdlib/blas/ext/base/dasumpw`
- [x] `@stdlib/blas/ext/base/dcusum`
- [x] `@stdlib/blas/ext/base/dcusumkbn`
- [x] `@stdlib/blas/ext/base/dcusumkbn2`
- [x] `@stdlib/blas/ext/base/dcusumors`
- [x] `@stdlib/blas/ext/base/dcusumpw`
- [x] `@stdlib/blas/ext/base/dfill`
- [x] `@stdlib/blas/ext/base/dnanasum`
- [x] `@stdlib/blas/ext/base/dnanasumors`
- [x] `@stdlib/blas/ext/base/dnannsum`
- [x] `@stdlib/blas/ext/base/dnannsumkbn`
- [x] `@stdlib/blas/ext/base/dnannsumkbn2`
- [x] `@stdlib/blas/ext/base/dnannsumors`
- [x] `@stdlib/blas/ext/base/dnannsumpw`
- [x] `@stdlib/blas/ext/base/dnansum`
- [x] `@stdlib/blas/ext/base/dnansumkbn`
- [x] `@stdlib/blas/ext/base/dnansumkbn2`
- [x] `@stdlib/blas/ext/base/dnansumors`
- [x] `@stdlib/blas/ext/base/dnansumpw`
- [x] `@stdlib/blas/ext/base/drev`
- [x] `@stdlib/blas/ext/base/dsapxsum`
- [x] `@stdlib/blas/ext/base/dsapxsumpw`
- [x] `@stdlib/blas/ext/base/dsnannsumors`
- [x] `@stdlib/blas/ext/base/dsnansum`
- [x] `@stdlib/blas/ext/base/dsnansumors`
- [x] `@stdlib/blas/ext/base/dsnansumpw`
- [x] `@stdlib/blas/ext/base/dsort2hp`
- [x] `@stdlib/blas/ext/base/dsort2ins`
- [x] `@stdlib/blas/ext/base/dsort2sh`
- [x] `@stdlib/blas/ext/base/dsorthp`
- [x] `@stdlib/blas/ext/base/dsortins`
- [x] `@stdlib/blas/ext/base/dsortsh`
- [x] `@stdlib/blas/ext/base/dssum`
- [x] `@stdlib/blas/ext/base/dssumors`
- [x] `@stdlib/blas/ext/base/dssumpw`
- [x] `@stdlib/blas/ext/base/dsum`
- [x] `@stdlib/blas/ext/base/dsumkbn`
- [x] `@stdlib/blas/ext/base/dsumkbn2`
- [x] `@stdlib/blas/ext/base/dsumors`
- [x] `@stdlib/blas/ext/base/dsumpw`
- [x] `@stdlib/blas/ext/base/sapx`
- [x] `@stdlib/blas/ext/base/sapxsum`
- [x] `@stdlib/blas/ext/base/sapxsumkbn`
- [x] `@stdlib/blas/ext/base/sapxsumkbn2`
- [x] `@stdlib/blas/ext/base/sapxsumors`
- [x] `@stdlib/blas/ext/base/sapxsumpw`
- [x] `@stdlib/blas/ext/base/sasumpw`
- [x] `@stdlib/blas/ext/base/scusum`
- [x] `@stdlib/blas/ext/base/scusumkbn`
- [x] `@stdlib/blas/ext/base/scusumkbn2`
- [x] `@stdlib/blas/ext/base/scusumors`
- [x] `@stdlib/blas/ext/base/scusumpw`
- [x] `@stdlib/blas/ext/base/sdsapxsum`
- [x] `@stdlib/blas/ext/base/sdsapxsumpw`
- [x] `@stdlib/blas/ext/base/sdsnansum`
- [x] `@stdlib/blas/ext/base/sdsnansumpw`
- [x] `@stdlib/blas/ext/base/sdssum`
- [x] `@stdlib/blas/ext/base/sdssumpw`
- [x] `@stdlib/blas/ext/base/sfill`
- [x] `@stdlib/blas/ext/base/snansum`
- [x] `@stdlib/blas/ext/base/snansumkbn`
- [x] `@stdlib/blas/ext/base/snansumkbn2`
- [x] `@stdlib/blas/ext/base/snansumors`
- [x] `@stdlib/blas/ext/base/snansumpw`
- [x] `@stdlib/blas/ext/base/srev`
- [x] `@stdlib/blas/ext/base/ssort2hp`
- [x] `@stdlib/blas/ext/base/ssort2ins`
- [x] `@stdlib/blas/ext/base/ssort2sh`
- [x] `@stdlib/blas/ext/base/ssorthp`
- [x] `@stdlib/blas/ext/base/ssortins`
- [x] `@stdlib/blas/ext/base/ssortsh`
- [x] `@stdlib/blas/ext/base/ssum`
- [x] `@stdlib/blas/ext/base/ssumkbn`
- [x] `@stdlib/blas/ext/base/ssumkbn2`
- [x] `@stdlib/blas/ext/base/ssumors`
- [x] `@stdlib/blas/ext/base/ssumpw`

### LINKED ISSUE #1509: [RFC]: refactor `blas/ext/base/sapx` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/sapx`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/sapx) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by rxbryan**
update blas/ext/base/sapx to follow current project conventions

## Description

 updates blas/ext/base/sapx to follow current project conventions as outlined in issue #788

This pull request:

-   resolves a part of #1152
-  fixes #1509
 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/sapx/README.md [hunk 1]
```diff
@@ -50,34 +50,30 @@ The function has the following parameters:
 -   **x**: input [`Float32Array`][@stdlib/array/float32].
 -   **stride**: index increment.
 
-The `N` and `stride` parameters determine which elements in `x` are accessed at runtime. For example, to add a constant to every other element
+The `N` and `stride` parameters determine which elements in strided array are accessed at runtime. For example, to add a constant to every other element
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/README.md]:
```suggestion
The `N` and `stride` parameters determine which elements in the strided array are accessed at runtime. For example, to add a constant to every other element
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/README.md [hunk 2]
```diff
@@ -98,7 +94,7 @@ The function has the following additional parameters:
 
 -   **offset**: starting index.
 
-While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the `offset` parameter supports indexing semantics based on a starting index. For example, to access only the last three elements of `x`
+While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the `offset` parameter supports indexing semantics based on a starting index. For example, to access only the last three elements of the strided array `x`
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/README.md]:
```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the `offset` parameter supports indexing semantics based on a starting index. For example, to access only the last three elements of the strided array
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/README.md [hunk 3]
```diff
@@ -117,7 +113,7 @@ sapx.ndarray( 3, 5.0, x, 1, x.length-3 );
 
 ## Notes
 
--   If `N <= 0`, both functions return `x` unchanged.
+-   If `N <= 0`, both functions return the strided array `x` unchanged.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/README.md]:
```suggestion
-   If `N <= 0`, both functions return the input array unchanged.
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/README.md [hunk 4]
```diff
@@ -117,7 +113,7 @@ sapx.ndarray( 3, 5.0, x, 1, x.length-3 );
 
 ## Notes
 
--   If `N <= 0`, both functions return `x` unchanged.
+-   If `N <= 0`, both functions return the input array unchanged.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/README.md:116]: Would you mind updating the example below, as well?

File: lib/node_modules/@stdlib/blas/ext/base/sapx/README.md [hunk 5]
```diff
@@ -50,34 +50,30 @@ The function has the following parameters:
...
 
 // Create an offset view...
 var x1 = new Float32Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
-var N = floor( x0.length/2 );
 
 // Add a constant to every other element...
-sapx( N, 5.0, x1, 2 );
+sapx( 3, 5.0, x1, 2 );
 // x0 => <Float32Array>[ 1.0, 3.0, 3.0, 1.0, 5.0, -1.0 ]
 ```
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/README.md:79]: Below: "...in a single-precision floating-point strided array using alternative indexing semantics."

File: lib/node_modules/@stdlib/blas/ext/base/sapx/benchmark/benchmark.native.js [hunk 6]
```diff
@@ -22,16 +22,17 @@
...
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float32Array = require( '@stdlib/array/float32' );
 var tryRequire = require( '@stdlib/utils/try-require' );
 var pkg = require( './../package.json' ).name;
 
 
 // VARIABLES //
 
+var rand = uniform( -100.0, 100.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/benchmark/benchmark.native.js]: Move this line below L39.

File: lib/node_modules/@stdlib/blas/ext/base/sapx/benchmark/benchmark.ndarray.native.js [hunk 7]
```diff
@@ -22,16 +22,17 @@
...
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float32Array = require( '@stdlib/array/float32' );
 var tryRequire = require( '@stdlib/utils/try-require' );
 var pkg = require( './../package.json' ).name;
 
 
 // VARIABLES //
 
+var rand = uniform( -100.0, 100.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/benchmark/benchmark.ndarray.native.js]: Move below L39.

File: lib/node_modules/@stdlib/blas/ext/base/sapx/manifest.json [hunk 8]
```diff
@@ -1,40 +1,46 @@
...
-				"./include"
-			],
-			"libraries": [
-				"-lm"
-			],
-			"libpath": [],
-			"dependencies": []
-		}
-	]
+    "options": {},
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/manifest.json]: Update to 2-space indentation.

File: lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/ext/base/sapx.h"
+#include <node_api.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c]: Move to last include to match include order convention of other packages.

File: lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+#include "stdlib/napi/export.h"
+#include "stdlib/napi/argv_double.h"
+#include "stdlib/napi/argv.h"
+#include "stdlib/napi/argv_int64.h"
+#include "stdlib/napi/argv_strided_float32array.h"
+
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c:27]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+* -   When called from JavaScript, the function expects four arguments:
+*
+*     -   `N`: number of indexed elements
+*     -   `alpha`: scalar
+*     -   `X`: input array
+*     -   `strideX`: `X` stride length
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
+	//napi_status status;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+
+	STDLIB_NAPI_ARGV_INT64(env, N, argv, 0);
+	STDLIB_NAPI_ARGV_DOUBLE(env, alpha, argv, 1);
+	STDLIB_NAPI_ARGV_INT64(env, strideX, argv, 3);
+ 
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY(env, X, N, strideX, argv, 2)
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c]:
```suggestion
	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
	STDLIB_NAPI_ARGV_DOUBLE( env, alpha, argv, 1 );
	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
 
	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 2 )
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+	c_sapx( N, (float)alpha, (float *)X, strideX );
+
+	return NULL;
+}
+
+STDLIB_NAPI_MODULE_EXPORT_FCN(addon)
+//NAPI_MODULE( NODE_GYP_MODULE_NAME, Init )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c]:
```suggestion
STDLIB_NAPI_MODULE_EXPORT_FCN( addon )
```

Don't leave in stray commented out code.

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks for working on this @rxbryan. This PR is missing updated examples and is missing updated tests.

**Comment by rxbryan** [comment]
> This PR is missing updated examples and is missing updated tests.

Thanks for the review @kgryte I will implement the changes ASAP

**Comment by rxbryan** [comment]
Hi @kgryte I have pushed the requested changes.

File: lib/node_modules/@stdlib/blas/ext/base/sapx/README.md (hunk 4 above)
Comment (rxbryan) [reply lib/node_modules/@stdlib/blas/ext/base/sapx/README.md:116]: I have updated the example to use utilities for generating the array

File: lib/node_modules/@stdlib/blas/ext/base/sapx/benchmark/benchmark.native.js (hunk 6 above)
Comment (rxbryan) [reply lib/node_modules/@stdlib/blas/ext/base/sapx/benchmark/benchmark.native.js]: I've made this change

File: lib/node_modules/@stdlib/blas/ext/base/sapx/benchmark/benchmark.ndarray.native.js (hunk 7 above)
Comment (rxbryan) [reply lib/node_modules/@stdlib/blas/ext/base/sapx/benchmark/benchmark.ndarray.native.js]: sorry for the styling mistakes but they are all fixed now 😅

File: lib/node_modules/@stdlib/blas/ext/base/sapx/manifest.json (hunk 8 above)
Comment (rxbryan) [reply lib/node_modules/@stdlib/blas/ext/base/sapx/manifest.json]: fixed.

**Comment by rxbryan** [comment]
Hi @kgryte  Sorry for the excessive styling mistakes but they are all fixed now. I have updated the examples and tests

File: lib/node_modules/@stdlib/blas/ext/base/sapx/README.md [hunk 14]
```diff
@@ -50,34 +50,30 @@ The function has the following parameters:
 -   **x**: input [`Float32Array`][@stdlib/array/float32].
 -   **stride**: index increment.
 
-The `N` and `stride` parameters determine which elements in `x` are accessed at runtime. For example, to add a constant to every other element
+The `N` and `stride` parameters determine which elements in the strided array are accessed at runtime. For example, to add a constant to every other element
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/README.md]:
```suggestion
The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to add a constant to every other element
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/README.md [hunk 15]
```diff
@@ -130,27 +126,13 @@ sapx.ndarray( 3, 5.0, x, 1, x.length-3 );
...
-    sign = randu();
-    if ( sign < 0.5 ) {
-        sign = -1.0;
-    } else {
-        sign = 1.0;
-    }
-    x[ i ] = sign * rand;
-}
+var rand = uniform( -100.0, 100.0 );
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/README.md [hunk 16]
```diff
@@ -130,27 +126,12 @@ sapx.ndarray( 3, 5.0, x, 1, x.length-3 );
...
-    sign = randu();
-    if ( sign < 0.5 ) {
-        sign = -1.0;
-    } else {
-        sign = 1.0;
-    }
-    x[ i ] = sign * rand;
-}
+var rand = uniform( -100.0, 100.0 );
+var x = filledarrayBy( 10, 'float32', rand );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/README.md]:
```suggestion
var x = filledarrayBy( 10, 'float32', uniform( -100.0, 100.0 ) );
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/examples/index.js [hunk 17]
```diff
@@ -18,27 +18,14 @@
...
+var filledarrayBy = require( '@stdlib/array/filled-by' );
 var sapx = require( './../lib' );
 
-var rand;
-var sign;
-var x;
-var i;
+var rand = uniform( -100.0, 100.0 );
+
+var x = filledarrayBy( 10, 'float32', rand );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/examples/index.js]:
```suggestion
var x = filledarrayBy( 10, 'float32', uniform( -100.0, 100.0 ) );
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/manifest.json [hunk 18]
```diff
@@ -1,40 +1,46 @@
...
-				"./include"
-			],
-			"libraries": [
-				"-lm"
-			],
-			"libpath": [],
-			"dependencies": []
-		}
-	]
+  "options": {},
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/manifest.json]: This file needs to be updated to include additional build configurations. See, e.g., https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/blas/ext/base/dfill/manifest.json

File: lib/node_modules/@stdlib/blas/ext/base/sapx/manifest.json [hunk 19]
```diff
@@ -1,40 +1,46 @@
...
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/napi/export",
+        "@stdlib/napi/argv",
+        "@stdlib/napi/argv-int64",
+        "@stdlib/napi/argv-double",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/manifest.json]:
```suggestion
        "@stdlib/napi/argv-float",
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/lib/ndarray.native.js [hunk 20]
```diff
@@ -47,10 +48,9 @@ var addon = require( './sapx.native.js' );
 */
 function sapx( N, alpha, x, stride, offset ) {
 	var view;
-	if ( stride < 0 ) {
-		offset += (N-1) * stride;
-	}
-	view = new Float32Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offset), x.length-offset ); // eslint-disable-line max-len
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/lib/ndarray.native.js]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+*
+* ## Notes
+*
+* -   When called from JavaScript, the function expects four arguments:
+*
+*     -   `N`: number of indexed elements
+*     -   `alpha`: scalar
+*     -   `X`: input array
+*     -   `strideX`: `X` stride length
+*/
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c]:
```suggestion
/**
* Receives JavaScript callback invocation data.
*
* @private
* @param env    environment under which the function is invoked
* @param info   callback data
* @return       Node-API value
*/
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_DOUBLE( env, alpha, argv, 1 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c]:
```suggestion
	STDLIB_NAPI_ARGV_FLOAT( env, alpha, argv, 1 );
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 ); 
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 2 )
+
+	c_sapx( N, (float)alpha, (float *)X, strideX );
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c]:
```suggestion
	stdlib_strided_sapx( N, alpha, X, strideX );
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_FLOAT( env, alpha, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 ); 
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 2 )
+	stdlib_strided_sapx( N, alpha, X, strideX );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c]:
```suggestion
	c_sapx( N, alpha, X, strideX );
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c [hunk 25]
```diff
@@ -0,0 +1,45 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/ext/base/sapx.h"
+#include "stdlib/napi/export.h"
+#include "stdlib/napi/argv.h"
+#include "stdlib/napi/argv_int64.h"
+#include "stdlib/napi/argv_double.h"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/src/addon.c]:
```suggestion
#include "stdlib/napi/argv_float.h"
```

**Review by kgryte — CHANGES_REQUESTED** [review]
The `manifest.json` file needs to be updated. And just checking: there were no changes necessary for `docs/repl.txt` or `docs/types/index.d.ts`?

**Comment by rxbryan** [comment]
> The `manifest.json` file needs to be updated. And just checking: there were no changes necessary for `docs/repl.txt` or `docs/types/index.d.ts`?

I've updated `manifest.json` and `docs/repl.txt`.  It did require some changes.

`docs/types/index.d.ts` looks okay

File: lib/node_modules/@stdlib/blas/ext/base/sapx/docs/repl.txt [hunk 26]
```diff
@@ -1,15 +1,15 @@
 
 {{alias}}( N, alpha, x, stride )
-    Adds a constant to each element in a single-precision floating-point strided
-    array.
+    Adds a constant to each element in a single-precision floating-point 
+    strided array.
 
-    The `N` and `stride` parameters determine which elements in `x` are accessed
-    at runtime.
+    The `N` and stride parameter determine which elements in the strided array
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/docs/repl.txt]:
```suggestion
    The `N` and stride parameters determine which elements in the strided array
```

File: lib/node_modules/@stdlib/blas/ext/base/sapx/docs/repl.txt [hunk 27]
```diff
@@ -1,15 +1,15 @@
 
 {{alias}}( N, alpha, x, stride )
-    Adds a constant to each element in a single-precision floating-point strided
-    array.
+    Adds a constant to each element in a single-precision floating-point 
+    strided array.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/docs/repl.txt]: Did this need to be rewrapped?

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/sapx/docs/repl.txt]: I'm surprised that L3 extended beyond 80 characters.

File: lib/node_modules/@stdlib/blas/ext/base/sapx/manifest.json [hunk 28]
```diff
@@ -1,40 +1,75 @@
...
-				"./include"
-			],
-			"libraries": [
-				"-lm"
-			],
-			"libpath": [],
-			"dependencies": []
-		}
-	]
+  "options": {},
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sapx/manifest.json]:
```suggestion
  "options": {
    "task": "build"
  },
```

Missing the default task.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @rxbryan!

**Pull request merged**